api/v1alpha4/azuremachinetemplate_webhook_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestAzureMachineTemplate_ValidateUpdate(t *testing.T) {
+	g := NewWithT(t)
+	failureDomain := "domaintest"
+
+	tests := []struct {
+		name        string
+		oldTemplate *AzureMachineTemplate
+		template    *AzureMachineTemplate
+		wantErr     bool
+	}{
+		{
+			name: "AzureMachineTemplate is immutable test1",
```

Comment:
can we have more meaningful names for test cases?

Reply:
A test case to make sure fields other than spec are mutable would be great.

Reply:
ok, I will perfect the test cases in next commit

Reply:
@mowangdk please update the test case names to be descriptive of what the test case is testing